packages/sitecore-jss/src/layout-service.ts

```diff
@@ -1,12 +1,12 @@
 import { AxiosDataFetcher } from './data-fetcher';
-import { LayoutServiceData } from './dataModels';
-import * as dataApi from './dataApi';
+import { LayoutServiceData, PlaceholderData } from './dataModels';
+import { fetchPlaceholderData, fetchRouteData, LayoutServiceConfig } from './dataApi';
 import { HttpJsonFetcher } from './httpClientInterface';
 import { IncomingMessage, ServerResponse } from 'http';
 
-export type DataFetcherResolver = (req?: IncomingMessage, res?: ServerResponse) => HttpJsonFetcher<LayoutServiceData>;
+export type DataFetcherResolver<T> = (req?: IncomingMessage, res?: ServerResponse) => HttpJsonFetcher<T>;
```

Comment:
Think we could make an exception and use `any` here to avoid having 2 DataFetcherResolvers on the `LayoutServiceInstanceConfig`.
```suggestion
export type DataFetcherResolver = (req?: IncomingMessage, res?: ServerResponse) => HttpJsonFetcher<any>;
```